### PR TITLE
[3.9] bpo-43077: Update bundled pip to 21.0.1 and setuptools to 52.0.0 (GH-24386)

### PR DESCRIPTION
**Pull request by pfmoore**
Update bundled pip to 21.0.1 and setuptools to 52.0.0.
(cherry picked from commit 4d11ecbb5ed78e6259ee27289c7638aad795f473)

Co-authored-by: Paul Moore <p.f.moore@gmail.com>

https://bugs.python.org/issue43077

### CODE REVIEW

**Comment by pfmoore** [comment]
:-( Looks like ensurepip changed between 3.9 and 3.10, probably to handle the fact that we're no longer shipping `py2.py3` wheels. I'll look at backporting that fix and hopefully that will address this.

File: Lib/ensurepip/__init__.py [hunk 1]
```diff
@@ -11,15 +11,12 @@
 
 
 __all__ = ["version", "bootstrap"]
-
-
-_SETUPTOOLS_VERSION = "49.2.1"
-
-_PIP_VERSION = "20.2.3"
-
+_PACKAGE_NAMES = ('setuptools', 'pip')
```
Comment (pradyunsg) [inline Lib/ensurepip/__init__.py:14]: Why is this being set

Comment (pfmoore) [reply Lib/ensurepip/__init__.py:14]: The PR didn't backport cleanly, and I had to manually fix the merge conflict. I didn't notice that this variable isn't used in the older version of the code. It's harmless, but can be fixed. It's *not* the cause of the test failures, sadly.

**Comment by pfmoore** [comment]
Test failures seem to be coming from a deprecation warning for a "legacy version" where the version being parsed is "pip". I have no clue why that's happening, but modifying the test (`test.test_venv`, line 450) to *not* treat warnings as errors causes the test to pass.

My tentative assumption is that something in the code has been incorrectly parsing "pip" as a version for some time, but it's been harmless until now. With legacy versions now being warned about, that causes the test to fail with a hard error here.

The bad news is I have no idea what is causing the code to try to parse "pip" as if it were a version. It's almost impossible to instrument the code, because it's being run directly from the wheel, and if I modify the code I'll need to regenerate `RECORD`, etc...

I don't have time right now to debug this much further. @pradyunsg if you have any insights, I'd be very grateful!

**Comment by pfmoore** [comment]
https://github.com/pypa/pip/issues/9540

**Comment by sbidoul** [comment]
Superseded by #25578

**Comment by matrixise** [comment]
Hi @pfmoore

I close this PR because it is superseded by #25578.

Thank you so much for your PR.

**Pull request closed without merging**